Subject: [PATCH] task 1.1.3 JDBC
---
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
--- a/pom.xml	(revision aef5c59360e285ffe59470ad4f8b0cf0edf0b093)
+++ b/pom.xml	(revision 31c761497b4401b3f2701f4ed717be5456c3d3cd)
@@ -21,7 +21,12 @@
     </build>
 
     <dependencies>
-<!--         Ваши зависимости-->
+        <!-- https://mvnrepository.com/artifact/org.postgresql/postgresql -->
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>42.7.5</version>
+        </dependency>
          <!-- https://mvnrepository.com/artifact/junit/junit -->
          <dependency>
              <groupId>junit</groupId>
Index: src/main/java/jm/task/core/jdbc/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jm/task/core/jdbc/Main.java b/src/main/java/jm/task/core/jdbc/Main.java
--- a/src/main/java/jm/task/core/jdbc/Main.java	(revision aef5c59360e285ffe59470ad4f8b0cf0edf0b093)
+++ b/src/main/java/jm/task/core/jdbc/Main.java	(revision 31c761497b4401b3f2701f4ed717be5456c3d3cd)
@@ -1,7 +1,25 @@
 package jm.task.core.jdbc;
 
+import jm.task.core.jdbc.dao.UserDaoJDBCImpl;
+
+
 public class Main {
     public static void main(String[] args) {
-        // реализуйте алгоритм здесь
+
+        UserDaoJDBCImpl dao = new UserDaoJDBCImpl();
+        dao.createUsersTable();
+        System.out.println();
+        dao.saveUser("Петя", "Бобров", (byte) 69);
+        dao.saveUser("Маша", "Сидорова", (byte) 52);
+        dao.saveUser("Евгений", "Михайлов", (byte) 41);
+        dao.saveUser("Дима", "Дубров", (byte) 19);
+        System.out.println();
+        dao.getAllUsers();
+        System.out.println();
+        dao.cleanUsersTable();
+        System.out.println();
+        dao.dropUsersTable();
+
+
     }
 }
Index: src/main/java/jm/task/core/jdbc/dao/UserDaoJDBCImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jm/task/core/jdbc/dao/UserDaoJDBCImpl.java b/src/main/java/jm/task/core/jdbc/dao/UserDaoJDBCImpl.java
--- a/src/main/java/jm/task/core/jdbc/dao/UserDaoJDBCImpl.java	(revision aef5c59360e285ffe59470ad4f8b0cf0edf0b093)
+++ b/src/main/java/jm/task/core/jdbc/dao/UserDaoJDBCImpl.java	(revision 31c761497b4401b3f2701f4ed717be5456c3d3cd)
@@ -1,7 +1,10 @@
 package jm.task.core.jdbc.dao;
 
 import jm.task.core.jdbc.model.User;
+import jm.task.core.jdbc.util.Util;
 
+import java.sql.*;
+import java.util.ArrayList;
 import java.util.List;
 
 public class UserDaoJDBCImpl implements UserDao {
@@ -10,26 +13,98 @@
     }
 
     public void createUsersTable() {
-
+        String sql = """
+                     CREATE TABLE IF NOT EXISTS users(
+                     id SERIAL PRIMARY KEY,
+                     name VARCHAR,
+                     lastname VARCHAR,
+                     age SMALLINT
+                  )
+                """;
+        try (Connection connection = Util.getConnection();
+             Statement statement = connection.createStatement()) {
+            System.out.println(statement.execute(sql));
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
     }
 
     public void dropUsersTable() {
-
+        String sql = "DROP TABLE users";
+        try (Connection connection = Util.getConnection();
+             Statement statement = connection.createStatement()) {
+            statement.execute(sql);
+            System.out.println("Таблица 'users' успешно удалена (или не существовала).");
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
     }
 
     public void saveUser(String name, String lastName, byte age) {
+        String sql = "INSERT INTO users(name, lastname, age) VALUES(?, ?, ?)";
+        try (Connection connection = Util.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setString(1, name);
+            preparedStatement.setString(2, lastName);
+            preparedStatement.setByte(3, age);
+            preparedStatement.execute();
+            System.out.println("User с именем - " + name + " добавлен в базу данных");
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
 
     }
 
     public void removeUserById(long id) {
+        String sql = "DELETE FROM users WHERE id = ?";
+        try (Connection connection = Util.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            int pe = preparedStatement.executeUpdate();
+
+            if (pe > 0) {
+                System.out.println("Пользователь с id = " + id + " удален из базы");
+            } else {
+                System.out.println("Пользователь с id = " + id + " не найден в базе");
+            }
+        } catch (SQLException e) {
+            System.out.println("Ошибка при удаление пользователя с id = " + id + ":");
+            e.printStackTrace();
+        }
 
     }
 
     public List<User> getAllUsers() {
-        return null;
+        List<User> users = new ArrayList<>();
+        String sql = "SELECT id, name, lastname, age FROM users";
+        try (Connection connection = Util.getConnection();
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(sql)) {
+            while (resultSet.next()) {
+                User user = new User();
+                user.setId(resultSet.getLong("id"));
+                user.setName(resultSet.getString("name"));
+                user.setLastName(resultSet.getString("lastname"));
+                user.setAge(resultSet.getByte("age"));
+                users.add(user);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        for (User user : users) {
+            System.out.println(user);
+        }
+        return users;
     }
 
     public void cleanUsersTable() {
-
+        String sql = "TRUNCATE  TABLE users";
+        try (Connection connection = Util.getConnection();
+             Statement statement = connection.createStatement()) {
+            statement.executeUpdate(sql);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
     }
 }
Index: src/main/java/jm/task/core/jdbc/model/User.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jm/task/core/jdbc/model/User.java b/src/main/java/jm/task/core/jdbc/model/User.java
--- a/src/main/java/jm/task/core/jdbc/model/User.java	(revision aef5c59360e285ffe59470ad4f8b0cf0edf0b093)
+++ b/src/main/java/jm/task/core/jdbc/model/User.java	(revision 31c761497b4401b3f2701f4ed717be5456c3d3cd)
@@ -22,6 +22,16 @@
 
     }
 
+    @Override
+    public String toString() {
+        return "User{" +
+               "id=" + id +
+               ", name='" + name + '\'' +
+               ", lastName='" + lastName + '\'' +
+               ", age=" + age +
+               '}';
+    }
+
     public User(String name, String lastName, Byte age) {
         this.name = name;
         this.lastName = lastName;
@@ -60,3 +70,4 @@
         this.age = age;
     }
 }
+
Index: src/main/java/jm/task/core/jdbc/service/UserService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jm/task/core/jdbc/service/UserService.java b/src/main/java/jm/task/core/jdbc/service/UserService.java
--- a/src/main/java/jm/task/core/jdbc/service/UserService.java	(revision aef5c59360e285ffe59470ad4f8b0cf0edf0b093)
+++ b/src/main/java/jm/task/core/jdbc/service/UserService.java	(revision 31c761497b4401b3f2701f4ed717be5456c3d3cd)
@@ -2,6 +2,7 @@
 
 import jm.task.core.jdbc.model.User;
 
+import java.sql.Connection;
 import java.util.List;
 
 public interface UserService {
Index: src/main/java/jm/task/core/jdbc/service/UserServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jm/task/core/jdbc/service/UserServiceImpl.java b/src/main/java/jm/task/core/jdbc/service/UserServiceImpl.java
--- a/src/main/java/jm/task/core/jdbc/service/UserServiceImpl.java	(revision aef5c59360e285ffe59470ad4f8b0cf0edf0b093)
+++ b/src/main/java/jm/task/core/jdbc/service/UserServiceImpl.java	(revision 31c761497b4401b3f2701f4ed717be5456c3d3cd)
@@ -1,31 +1,45 @@
 package jm.task.core.jdbc.service;
 
+
+import jm.task.core.jdbc.dao.UserDaoJDBCImpl;
 import jm.task.core.jdbc.model.User;
 
 import java.util.List;
 
+
 public class UserServiceImpl implements UserService {
+    UserDaoJDBCImpl userDao = new UserDaoJDBCImpl();
+
+    @Override
     public void createUsersTable() {
-
+        userDao.createUsersTable();
     }
 
+    @Override
     public void dropUsersTable() {
-
+        userDao.dropUsersTable();
     }
 
+    @Override
     public void saveUser(String name, String lastName, byte age) {
+        userDao.saveUser(name, lastName, age);
 
     }
 
+    @Override
     public void removeUserById(long id) {
+        userDao.removeUserById(id);
 
     }
 
+    @Override
     public List<User> getAllUsers() {
-        return null;
+        return userDao.getAllUsers();
     }
 
+    @Override
     public void cleanUsersTable() {
+        userDao.cleanUsersTable();
 
     }
 }
Index: src/main/java/jm/task/core/jdbc/util/Util.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jm/task/core/jdbc/util/Util.java b/src/main/java/jm/task/core/jdbc/util/Util.java
--- a/src/main/java/jm/task/core/jdbc/util/Util.java	(revision aef5c59360e285ffe59470ad4f8b0cf0edf0b093)
+++ b/src/main/java/jm/task/core/jdbc/util/Util.java	(revision 31c761497b4401b3f2701f4ed717be5456c3d3cd)
@@ -1,5 +1,23 @@
 package jm.task.core.jdbc.util;
 
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
 public class Util {
+
     // реализуйте настройку соеденения с БД
-}
+    private static String url = "jdbc:postgresql://localhost:5432/postgres";
+    private static String user = "postgres";
+    private static String password = "zadedi92";
+
+    public static Connection getConnection() throws SQLException {
+        Connection connection = DriverManager.getConnection(url, user, password);
+            if (connection != null) {
+                System.out.println("Подключение к postgreSQL");
+            } else {
+                System.out.println("Ошибка подключения к postgreSQL");
+            }
+            return connection;
+        }
+    }
